app/webpacker/components/ContactEditProfilePage/index.jsx

```diff
@@ -8,31 +8,88 @@ import Loading from '../Requests/Loading';
 import Errored from '../Requests/Errored';
 import EditProfileForm from './EditProfileForm';
+import useLoggedInUserPermissions from '../../lib/hooks/useLoggedInUserPermissions';
+import useQueryParams from '../../lib/hooks/useQueryParams';
+import useInputState from '../../lib/hooks/useInputState';
+import WcaSearch from '../SearchWidget/WcaSearch';
+import SEARCH_MODELS from '../SearchWidget/SearchModel';
 
 const CONTACT_EDIT_PROFILE_QUERY_CLIENT = new QueryClient();
 
-export default function ContactEditProfilePage({ loggedInUserId, recaptchaPublicKey }) {
+function ContactEditProfileSelf({ setContactSuccess, recaptchaPublicKey }) {
   const { data: loggedInUserData, isLoading, isError } = useQuery({
     queryKey: ['userData'],
     queryFn: () => fetchJsonOrError(apiV0Urls.users.me.userDetails),
-    enabled: !!loggedInUserId,
   }, CONTACT_EDIT_PROFILE_QUERY_CLIENT);
   const wcaId = loggedInUserData?.data?.user?.wca_id;
-  const [contactSuccess, setContactSuccess] = useState(false);
 
   if (isLoading) return <Loading />;
   if (isError) return <Errored />;
-  if (!loggedInUserData) {
+  if (!wcaId) {
     return (
       <Message error>
-        <I18nHTMLTranslate i18nKey="page.contact_edit_profile.not_logged_in_error" />
+        <I18nHTMLTranslate i18nKey="page.contact_edit_profile.no_profile_error" />
       </Message>
     );
   }
-  if (loggedInUserData && !wcaId) {
+
+  return (
+    <Container text>
+      <Header as="h2">{I18n.t('page.contact_edit_profile.title')}</Header>
+      <EditProfileForm
+        wcaId={wcaId}
+        onContactSuccess={() => setContactSuccess(true)}
+        recaptchaPublicKey={recaptchaPublicKey}
+      />
+    </Container>
+  );
+}
+
+function ContactEditProfileOthers({ setContactSuccess, recaptchaPublicKey }) {
+  const { loggedInUserPermissions, loading } = useLoggedInUserPermissions();
+  const [person, setPerson] = useInputState();
+  const wcaId = person?.id;
+
+  if (loading) return <Loading />;
+
+  if (!loggedInUserPermissions.canRequestToEditOthersProfile) {
     return (
       <Message error>
-        <I18nHTMLTranslate i18nKey="page.contact_edit_profile.no_profile_error" />
+        <I18nHTMLTranslate i18nKey="page.contact_edit_profile.no_permission_error" />
+      </Message>
+    );
+  }
+
+  return (
+    <Container text>
+      <Header as="h2">{I18n.t('page.contact_edit_profile.title')}</Header>
+      <WcaSearch
+        model={SEARCH_MODELS.person}
+        multiple={false}
+        value={person}
+        onChange={setPerson}
+        disabled={!!wcaId}
+        label={I18n.t('page.contact_edit_profile.form.wca_id_search.label')}
+      />
+      {wcaId && (
+        <EditProfileForm
+          wcaId={wcaId}
+          onContactSuccess={() => setContactSuccess(true)}
+          recaptchaPublicKey={recaptchaPublicKey}
+        />
+      )}
+    </Container>
+  );
+}
+
+export default function ContactEditProfilePage({ loggedInUserId, recaptchaPublicKey }) {
```

Comment:
It's nice to have the main/exported component at the top, and the helper components below.

Reply:
Okay changed, I actually thought the reverse - like the exported component should be at bottom - just like how we have main method at bottom of programs.

Reply:
I think that's often the case because in some languages anything that a main method references has to have already been defined, so you have to put it at the bottom. Which isn't the case with React.

The point of this file is to provide (export) a component, so the most useful thing to see first is that component. You can then scroll down further to see more details if you want.

But it's not a big deal.